src/App.php

```diff
@@ -337,18 +342,19 @@ public function getBasePath(): string
 	 * @param IManagePersonalConfigValues $pConfig  Personal configuration
 	 * @param IHandleSessions             $session  The Session handler
 	 */
-	public function __construct(Database $database, IManageConfigValues $config, App\Mode $mode, BaseURL $baseURL, LoggerInterface $logger, Profiler $profiler, L10n $l10n, Arguments $args, IManagePersonalConfigValues $pConfig, IHandleSessions $session)
+	public function __construct(Database $database, IManageConfigValues $config, App\Mode $mode, BaseURL $baseURL, LoggerInterface $logger, Profiler $profiler, L10n $l10n, Arguments $args, IManagePersonalConfigValues $pConfig, IHandleSessions $session, IHandleUserSessions $userSession)
 	{
-		$this->database = $database;
-		$this->config   = $config;
-		$this->mode     = $mode;
-		$this->baseURL  = $baseURL;
-		$this->profiler = $profiler;
-		$this->logger   = $logger;
-		$this->l10n     = $l10n;
-		$this->args     = $args;
-		$this->pConfig  = $pConfig;
-		$this->session  = $session;
+		$this->database    = $database;
+		$this->config      = $config;
+		$this->mode        = $mode;
+		$this->baseURL     = $baseURL;
+		$this->profiler    = $profiler;
+		$this->logger      = $logger;
+		$this->l10n        = $l10n;
+		$this->args        = $args;
+		$this->pConfig     = $pConfig;
+		$this->session     = $session;
+		$this->userSession = $userSession;
```

Comment:
Three options I can think about:
- `UserSession` inherits `Session` --> this feels for me like we would build unnecessary dependency-trees
- `UserSession` contains `Session` and a method `getSession()` --> maybe?
- we leave it as is :)

Reply:
I would find it normal to perform `UserSession->get` to get regular session keys but then the existence of `Session` is put into question, which puts `UserSession`'s creation into question.

My main point of contention is that event with a `UserSession` object we can't be sure a user is logged in, which makes it to me semantically equivalent to a regular `Session`. The issue was that `Core\Session` was static while `UserSession` is preferably dynamic, and that we need session adapters for tests.

But I feel like this can be resolved into a single class.

Reply:
I'm still not sure to mix the "basic" `Session` context (= CRUD of Session) with "advanced" methods, which are having business logic as well.. But I think at first I will make this PRs finished and than I will merge `Session` with `UserSession` again :D

Reply:
Ideally we shouldn't query session keys directly. Everything should be behind named values like the `getLocalUserId` method.